cxplat/src/cxplat_winuser/cxplat_winuser.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+// This file contains initialization/cleanup routines for the Windows user-mode cxplat library.
+#include "cxplat.h"
+#include "leak_detector.h"
+#include "symbol_decoder.h"
+#include <algorithm>
+#include <memory>
+#include <string>
+
+extern "C" bool cxplat_fuzzing_enabled = false;
+
+cxplat_leak_detector_ptr _cxplat_leak_detector_ptr;
+
+/**
+ * @brief Environment variable to enable fault injection testing.
+ */
+#define CXPLAT_FAULT_INJECTION_SIMULATION_ENVIRONMENT_VARIABLE_NAME "CXPLAT_FAULT_INJECTION_SIMULATION"
+#define CXPLAT_MEMORY_LEAK_DETECTION_ENVIRONMENT_VARIABLE_NAME "CXPLAT_MEMORY_LEAK_DETECTION"
+
+/**
+ * @brief Get an environment variable as a string.
+ *
+ * @param[in] name Environment variable name.
+ * @return String value of environment variable or an empty string if not set.
+ */
+static std::string
+_get_environment_variable_as_string(const std::string& name)
+{
+    std::string value;
+    size_t required_size = 0;
+    getenv_s(&required_size, nullptr, 0, name.c_str());
+    if (required_size > 0) {
+        value.resize(required_size);
+        getenv_s(&required_size, &value[0], required_size, name.c_str());
+        value.resize(required_size - 1);
+    }
+    return value;
+}
+
+/**
+ * @brief Get an environment variable as a boolean.
+ *
+ * @param[in] name Environment variable name.
+ * @retval false Environment variable is set to "false", "0", or if it's not set.
+ * @retval true Environment variable is set to any other value.
+ */
+static bool
+_get_environment_variable_as_bool(const std::string& name)
+{
+    std::string value = _get_environment_variable_as_string(name);
+    if (value.empty()) {
+        return false;
+    }
+
+    // Convert value to lower case.
+    std::transform(value.begin(), value.end(), value.begin(), [](unsigned char c) { return (char)std::tolower(c); });
+    if (value == "false") {
+        return false;
+    }
+    if (value == "0") {
+        return false;
+    }
+    return true;
+}
+
+/**
+ * @brief Get an environment variable as a size_t.
+ *
+ * @param[in] name Environment variable name.
+ * @return Value of environment variable or 0 if it's not set or not a valid number.
+ */
+static size_t
+_get_environment_variable_as_size_t(const std::string& name)
+{
+    std::string value = _get_environment_variable_as_string(name);
+    if (value.empty()) {
+        return 0;
+    }
+    try {
+        return std::stoull(value);
+    } catch (const std::exception&) {
+        return 0;
+    }
+}
+
+cxplat_status_t
+cxplat_initialize()
+{
+    try {
+        auto fault_injection_stack_depth =
```

Comment:
Eventually, this all needs to be DEBUG only.

Reply:
Filed #86